unstable/wlr-export-dmabuf-unstable-v1.xml

```diff
@@ -0,0 +1,201 @@
+      </description>
+      <arg name="tv_sec_hi" type="uint"
+           summary="high 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_sec_lo" type="uint"
+           summary="low 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_nsec" type="uint"
+           summary="nanoseconds part of the timestamp"/>
+    </event>
+
+    <enum name="cancel_reason">
+      <description summary="cancel reason">
+        Indicates reason for cancelling the frame.
+      </description>
+      <entry name="temporary" value="0"
+             summary="temporary error, source will produce more frames"/>
+      <entry name="permanent" value="1"
+             summary="fatal error, source will not produce frames"/>
+      <entry name="resizing" value="2"
+             summary="temporary error, source will produce frames"/>
+    </enum>
+
+    <event name="cancel">
+      <description summary="indicates the frame is no longer valid">
+        If the frame is no longer valid after the "frame" event has been called,
+        this callback will be used to inform the client to scrap the frame.
+        Source is still valid for as long as the subscription function does not
+        return NULL.
+        This may get called if for instance the surface is in the process of
+        resizing.
+      </description>
+      <arg name="reason" type="uint" enum="cancel_reason"
+           summary="indicates a reason for cancelling this frame capture"/>
+    </event>
+
+    <request name="destroy" type="destructor">
+      <description summary="delete this object, used or not">
+        Unreferences the frame, allowing it to be reused. Must be called as soon
+        as its no longer used.
+        Can be called at any time by the client after the "frame" event, after
+        which the compositor will not call any other events unless the client
+        resubscribes to capture more. The client will still have to close any
+        FDs it has been given.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_export_dmabuf_manager_v1" version="1">
+    <description summary="manager to inform clients and begin capturing">
+      This object is a manager with which to start capturing from sources.
+    </description>
+
+    <request name="capture_output">
+      <description summary="subscribe to start capturing">
+        Request to start capturing from an entire wl_output.
+      </description>
+      <arg name="frame" type="new_id"
+           interface="zwlr_export_dmabuf_frame_v1"/>
+      <arg name="overlay_cursor" type="int"
+           summary="include custom client hardware cursor on top of the frame"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </request>
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
```

Comment:
Typically it's preferred to make destroying the parent interface a protocol error if any of its child resources are still alive

Reply:
Hmm, not really. For instance the linux-dmabuf protocol also allows this. Same for wl_compositor, wl_subcompositor…

Reply:
The example I was looking at is xdg-shell. But if there's no consensus then forget it, this is fine.